src/status_im/utils/async.cljs

```diff
@@ -0,0 +1,25 @@
+(ns status-im.utils.async
+  "Utility namespace containing `core.async` helper constructs"
+  (:require-macros [cljs.core.async.macros :as async])
+  (:require [cljs.core.async :as async]))
+
+(defn chunked-pipe!
+  "Connects input channel to the output channel with time-based chunking.
+  `flush-time` parameter decides for how long we are waiting to accumulate
+  value from input channel in a vector before it's put on the output channel.
+  When `flush-time` interval elapses and there are no values accumulated, nothing
+  is put on the output channel till the next input arrives, which is then put on
+  the output channel immediately (wrapped in a vector).
+  When input channel is closed, output channel is closed as well and go-loop exits."
+  [input-ch output-ch flush-time]
+  (async/go-loop [acc []
+                  flush false]
+    (if (and flush (seq acc))
+      (do (async/put! output-ch acc)
+          (recur [] false))
+      (let [[v ch] (async/alts! [input-ch (async/timeout flush-time)])]
+        (if (= ch input-ch)
+          (if v
+            (recur (conj acc v) flush)
+            (async/close! output-ch))
+          (recur acc true))))))
```

Comment:
Couldn't flush here be called `flush?` and set to `(seq acc)`? That way first statement would read `if flush?`

Reply:
No, `(seq acc)` as a sole condition for putting chunk on the output channel wouldn't work, because when timeout has not expired yet, we still want to wait for (potential) next input to the chunk.
But I agree that `flush?` is better more idiomatic name for the binding, so I will rename it.

Reply:
I don't understand, isn't it the same logic just more clear semantically? Sketch, not tested:

```
(defn chunked-pipe!
  (async/go-loop [acc []
                  flush? false]
     (if flush?
       (do (async/put! output-ch acc)
           (recur [] false))
       (let [[v ch] (async/alts! [input-ch (async/timeout flush-time)])]
         (if (= ch input-ch)
          (if v
            (recur (conj acc v) (and flush? (seq acc)))
             (async/close! output-ch))
           (recur acc (seq acc)))))))
```

Reply:
I misunderstood you, yes, that's exactly the same, just the `(and flush? (seq acc))` expression is moved from the first `if` to the `recur`.
I agree that's bit clearer, will change it.